### PR TITLE
load-fragment: parse specifiers in CPUAffinity=

### PR DESCRIPTION
**Pull request by fgervais**
We have that use-case where we have multiple hardware watchdogs which are meant to be bound 1 per CPU core.

So we use this service template:

```
[Unit]
Description=Hardware watchdog manager

[Service]
CPUAffinity=%i
ExecStart=/usr/sbin/watchdog -T 120 -F /dev/watchdog%i

[Install]
WantedBy=multi-user.target
```

and we make these instances:

```
watchdog@0.service
watchdog@1.service
watchdog@2.service
watchdog@3.service
```

This PR is here to make the `%i` work in the `CPUAffinity=` parameter.

### CODE REVIEW

File: src/core/load-fragment.c [hunk 1]
```diff
@@ -1697,6 +1697,8 @@ int config_parse_exec_cpu_affinity(
                 void *userdata) {
 
         ExecContext *c = data;
+        const Unit *u = userdata;
+        char *k;
```
Comment (yuwata) [inline src/core/load-fragment.c]: `k` is not freed. Please use `_cleanup_free_` attribute:
```
_cleanup_free_ chra *k = NULL;
```

File: src/core/load-fragment.c [hunk 2]
```diff
@@ -1711,7 +1713,15 @@ int config_parse_exec_cpu_affinity(
                 return 0;
         }
 
-        r = parse_cpu_set_extend(rvalue, &c->cpu_set, true, unit, filename, line, lvalue);
+        r = unit_full_printf(u, rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to resolve unit specifiers in '%s'",
+                           rvalue);
+                return -ENOEXEC;
```
Comment (yuwata) [inline src/core/load-fragment.c]: I do not think this should be critical. Please downgrade the log level to `LOG_WARNING`, add `", ignoring"` in the log message, and `return 0;`:
```
log_syntax(unit, LOG_WARNING, filename, line, r, "Failed to resolve unit specifiers in '%s', ignoring: %m");
return 0;
```

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/core/load-fragment.c (hunk 1 above)
Comment (fgervais) [reply src/core/load-fragment.c]: Ready, thank you

File: src/core/load-fragment.c (hunk 2 above)
Comment (fgervais) [reply src/core/load-fragment.c]: Ready, thank you

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**